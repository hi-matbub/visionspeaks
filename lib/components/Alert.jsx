import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ status, children }) => (
  <div style={{ height: '100px', width: '300px' }} className="mt-2">
    {status ? (
      <div className={`alert alert-${status}`} role="alert">
        {children}
      </div>
    ) : null}
  </div>
);

Alert.propTypes = {
  status: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Alert;
