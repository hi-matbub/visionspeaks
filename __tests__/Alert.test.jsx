import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Alert from '../lib/components/Alert';

const alert = {
  status: 'danger',
  message: 'Unknown error occured. Try again later.',
};

it('Alert component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Alert status={alert.status}>{alert.message}</Alert>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Alert component renders unchanged', () => {
const tree = renderer.create(<Alert status={alert.status}>{alert.message}</Alert>).toJSON();
  expect(tree).toMatchSnapshot();
});
