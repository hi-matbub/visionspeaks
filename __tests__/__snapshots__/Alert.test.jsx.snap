// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders unchanged 1`] = `
<div
  className="mt-2"
  style={
    Object {
      "height": "100px",
      "width": "300px",
    }
  }
>
  <div
    className="alert alert-danger"
    role="alert"
  >
    Unknown error occured. Try again later.
  </div>
</div>
`;
